core/codegen/tests/from_form.rs

```diff
@@ -612,3 +614,242 @@ fn test_multipart() {
 
     assert!(response.status().class().is_success());
 }
+
+fn test_hashmap() -> HashMap<&'static str, &'static str> {
+    let mut map = HashMap::new();
+    map.insert("key", "value");
+    map
+}
+
+fn test_btreemap() -> BTreeMap<&'static str, &'static str> {
+    let mut map = BTreeMap::new();
+    map.insert("key", "value");
+    map
+}
+
+mod default_macro_tests {
```

Comment:
We should move the unique cases in this `mod` into the test and check that the default is actually used. An important missing one is to check that `None` works as expected, especially for `bool`.